src/server.rs

```diff
@@ -379,7 +379,7 @@ fn determine_compressor(accept_encoding: String) -> Option<CompressionType> {
 
 	// This loop reads the requested compressors and keeps track of whichever
 	// one has the highest priority per our heuristic.
-	for val in accept_encoding.to_string().split(',') {
+	for val in accept_encoding.split(',') {
```

Comment:
I'm not opposed to this, but in the future, let's raise small fixes like these in a separate deliverable.

Reply:
You are right, this one is outside of the scope of this PR, not sure why I staged that.